src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/PackageWriter.java

```diff
@@ -275,17 +271,15 @@ private List<PackageElement> findRelatedPackages() {
 
     @Override
     protected Navigation getNavBar(PageMode pageMode, Element element) {
-        Content linkContent = getModuleLink(utils.elementUtils.getModuleOf(packageElement),
-                contents.moduleLabel);
-        return super.getNavBar(pageMode, element)
-                .setNavLinkModule(linkContent)
-                .setSubNavLinks(() -> List.of(
-                        links.createLink(HtmlIds.PACKAGE_DESCRIPTION, contents.navDescription,
-                                !utils.getFullBody(packageElement).isEmpty() && !options.noComment()),
-                        links.createLink(HtmlIds.RELATED_PACKAGE_SUMMARY, contents.relatedPackages,
-                                relatedPackages != null && !relatedPackages.isEmpty()),
-                        links.createLink(HtmlIds.CLASS_SUMMARY, contents.navClassesAndInterfaces,
-                                allClasses != null && !allClasses.isEmpty())));
+        List<Content> subnavLinks = new ArrayList<>();
+        if (configuration.showModules) {
+            ModuleElement mdle = configuration.docEnv.getElementUtils().getModuleOf(packageElement);
+            subnavLinks.add(links.createLink(pathToRoot.resolve(docPaths.moduleSummary(mdle)),
+                    Text.of(mdle.getQualifiedName())));
+        }
```

Comment:
This code sequence is showing up fairly often. Should it be shared somewhere for reuse?  For example, a method on a common super type?